webapp/components/user_settings/user_settings_security.jsx

```diff
@@ -73,6 +73,15 @@ class SecurityTab extends React.Component {
         };
     }
 
+    componentWillUpdate() {
+        if ((this.state.passwordError !== '' || this.state.serverError) && (this.state.currentPassword !== '' || this.state.newPassword !== '' || this.state.confirmPassword !== '')) {
```

Comment:
You aren't supposed to be able to use setState in componentWillUpdate anyway. I'd just clear the errors in the submit function before any other work is done. That's how [other components](https://github.com/mattermost/platform/blob/master/webapp/components/backstage/add_incoming_webhook.jsx#L43) do it.